### PR TITLE
Add lambda for more deatiled build status checking

### PR DESCRIPTION
**Pull request by fredemmott**
Example usage (intentionally an unauthenticated API):

```
fredemmott-pro:hhvm-packaging fredemmott$ curl https://3gmrqerxm6.execute-api.us-west-2.amazonaws.com/prod/build-status/3.27.2 | jq
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   243  100   243    0     0     82      0  0:00:02  0:00:02 --:--:--    82
{
  "success": true,
  "version": "3.27.2",
  "succeeded": [
    "macos-sierra",
    "macos-high_sierra",
    "source",
    "source_gpg",
    "debian-8-jessie",
    "debian-9-stretch",
    "ubuntu-14.04-trusty",
    "ubuntu-16.04-xenial",
    "ubuntu-17.10-artful",
    "ubuntu-18.04-bionic"
  ],
  "failed": {}
}
fredemmott-pro:hhvm-packaging fredemmott$ curl https://3gmrqerxm6.execute-api.us-west-2.amazonaws.com/prod/build-status/nightly | jq
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   344  100   344    0     0    363      0 --:--:-- --:--:-- --:--:--   363
{
  "success": false,
  "version": "2018.08.20",
  "succeeded": [
    "macos-sierra",
    "macos-high_sierra",
    "source",
    "source_gpg",
    "debian-8-jessie",
    "debian-9-stretch",
    "ubuntu-16.04-xenial",
    "ubuntu-17.10-artful",
    "ubuntu-18.04-bionic"
  ],
  "failed": {
    "ubuntu-14.04-trusty": "https://dl.hhvm.com/ubuntu/pool/main/h/hhvm-nightly/hhvm-nightly_2018.08.20-1~trusty_amd64.deb"
  }
}
```

### CODE REVIEW

File: aws/lambdas/hhvm-get-build-status/index.js [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+async function get_distros(branch) {
+  const response = await rp(get_distros_uri(branch));
+  return response.trim().split("\n");
+}
+
+exports.handler = async (event) => {
+  const {nightly, version, branch} = get_version_info(event);
+
+  const prefix = nightly ? 'hhvm-nightly-'+version : 'hhvm-'+version;
+  let paths = {
```
Comment (kmeht) [inline aws/lambdas/hhvm-get-build-status/index.js]: Can be `const`, you just can't rebind `paths`.

File: aws/lambdas/hhvm-get-build-status/index.js [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+  if (nightly) {
+    paths.source = 'source/nightlies/'+prefix+'.tar.gz';
+    paths.source_gpg = 'source/nightlies/'+prefix+'.tar.gz.sig';
+  } else {
+    paths.source = 'source/'+prefix+'.tar.gz';
+    paths.source_gpg = 'source/'+prefix+'.tar.gz.sig';
+  }
+
+  const distros = await get_distros(branch);
+  distros.forEach(distro => {
```
Comment (kmeht) [inline aws/lambdas/hhvm-get-build-status/index.js]: Can't use `for...of`?

File: aws/lambdas/hhvm-get-build-status/index.js [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+        version + '-1~' + (distro.match(/[a-z]+$/)) + '_amd64.deb';
+      return;
+    }
+    // If we add a new distro kind without updating monitoring, make
+    // the monitoring fail.
+    paths[distro] = distro;
+  });
+
+  let success = true;
+  let results = { };
```
Comment (kmeht) [inline aws/lambdas/hhvm-get-build-status/index.js]: nit: no space

File: aws/lambdas/hhvm-get-build-status/index.js [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+  const s3 = new AWS.S3();
+  await promise.all(
+    Object.values(paths).map(path => (async function() {
```
Comment (kmeht) [inline aws/lambdas/hhvm-get-build-status/index.js]: Do we have to do it this way? vs. something like
```
Object.values(paths).map(async (path) => {
  //...
});
```

File: aws/lambdas/hhvm-get-build-status/index.js (hunk 2 above)
Comment (fredemmott) [reply aws/lambdas/hhvm-get-build-status/index.js]: TIL

File: aws/lambdas/hhvm-get-build-status/index.js (hunk 4 above)
Comment (fredemmott) [reply aws/lambdas/hhvm-get-build-status/index.js]: there's `()` at the end - I'm returning an Awaitable, not a (function():Awaitable)

**Pull request merged**